### PR TITLE
resource-consumer windows image use leak-and-touch for memory allocations

### PR DESCRIPTION
**Pull request by marosset**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

This PR updates the type of memory allocation performed by the `resource-consumer` test image on Windows which is used for various resource limits / scaling tests.

These updates are being made to:

-  Address some issues in new HPA tests (discussed in [slack]( https://kubernetes.slack.com/archives/C0SJ4AFB7/p1664236027305249?thread_ts=1663957525.118749&cid=C0SJ4AFB7))
- To align with integration tests for container stat reporting in containerd project https://github.com/containerd/containerd/blob/8b804c4f817d63347406f4cfcefb376532f9eaeb/integration/container_stats_test.go#L117-L120

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/images/resource-consumer/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.11
+1.13
```
Comment (marosset) [inline test/images/resource-consumer/VERSION:1]: The version bump to 1.13 is intentional.
I'd like to get https://github.com/kubernetes/kubernetes/pull/112924 merged first which bumps image version to 1.12

**Comment by marosset** [comment]
/hold 
for https://github.com/kubernetes/kubernetes/pull/112924

**Comment by marosset** [comment]
/hold cancel
Since the other PR merged.

**Comment by claudiubelu** [comment]
Looks good, but the resource-consumer image for 1.12 was not promoted yet. In fact, the image building job for it failed (https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kubernetes-push-e2e-resource-consumer-test-images/1580625891615051776). Looking into it.

We should merge this after the 1.12 image is built and promoted, IMO.

/hold

**Comment by claudiubelu** [comment]
The resource-consumer:1.12 image has been built and it is currently proposed for promotion here: https://github.com/kubernetes/k8s.io/pull/4416

Because the resource-consumer:1.12 was built, we can go ahead with this PR.

/lgtm
/approve
/hold cancel

File: test/images/resource-consumer/VERSION (hunk 1 above)
Comment (dims) [reply test/images/resource-consumer/VERSION:1]: looks like that has merged. good!

**Pull request merged**